src/devices/src/virtio/queue.rs

```diff
@@ -161,6 +165,44 @@ impl<'a> DescriptorChain<'a> {
             None
         }
     }
+
+    // Gets a list of writtable buffers contained in this descriptor chain.
+    // Returns an error if descriptor chain is malformed.
+    pub fn get_wo_buffs(
```

Comment:
This logic seems specific to the net device. I wouldn't define it in the Queue implementation. An option would be to create a separate struct that contains the net queues and maybe the rate limiters or other components (`VirtioFrontend` for example) and define it there. Although this change would be quite big. Or we can simply define this method in the net device.